### DISCUSSION TITLE
Extract useful Fabric proto utils into a separate Go module

### DISCUSSION THREAD

**Discussion by jt-nti** [hyperledger/fabric#4608] [General]
From @davidkhala hyperledger/fabric-protos-go#10

> It is a awesome direction that we split protobuf out from fabric itself.
Meanwhie in second development, fabric-protos-go is used frequently along with powerful tool github.com/hyperledger/fabric/protoutil
Would you consider also migrate github.com/hyperledger/fabric/protoutil to somewhere in github.com/hyperledger/fabric-protos-go

Perhaps [fabric-lib-go](https://github.com/hyperledger/fabric-lib-go) would be a good home, and some thought about what to move is probably required since protoutil is currently a bit of a mixed bag.

**Comment by bestbeforetoday** [comment]
Any capability extracted out to a separate repository should probably also operate on [fabric-protos-go-apiv2](https://github.com/hyperledger/fabric-protos-go-apiv2), not on [fabric-protos-go](https://github.com/hyperledger/fabric-protos-go).

**Comment by jt-nti** [comment]
It would definitely be preferable to use apiv2 but I think that depends on where/when it gets extracted, and what consumes the separate module, e.g. fabric-lib-go would probably need to use the old bindings until other components catch up.

**Comment by davidkhala** [comment]
@jt-nti @bestbeforetoday Any progres on this issue? Or did I miss something great?

**Comment by bestbeforetoday** [comment]
As far as I know, nobody is working on this. Somebody else might have better information.

**Comment by davidkhala** [comment]
@jt-nti @bestbeforetoday I have some free time to pick up this task recently, starting from here https://github.com/davidkhala/fabric-protoutil/, I copy the protoutil package to here and trying to isolating it from fabric core as much as possible. 

Welcome comment on direction 
@denyeart Dave, would you mind moving this to a github discussion?

**Comment by denyeart** [comment]
@davidkhala 

I've moved this to a Discussion.

I agree in concept about moving protoutil to fabric-lib-go, that's the intended place for shared common code. The challenge is that it will pull in some other fabric dependencies. If you can manage that in a PR for review it would be appreciated.

There is separate discussion about moving fabric repositories to APIv2 in https://github.com/hyperledger/fabric/issues/3650. I'd suggest to keep the two work items independent. Which one would you like to attempt first @davidkhala ?

**Reply by davidkhala** [reply]
sure I will start. @denyeart I prefer to start fabric-lib-go migration first. I guess that is relatively smaller in effort.

https://github.com/hyperledger/fabric-lib-go/pull/23
